File: Sorting/selection_sort.py
def selection_sort(arr):
    for i in range(len(arr)-1):
        minIndex = i
        for j in range(i+1,len(arr)):
            if(arr[j] < arr[minIndex]): 
                minIndex = j
        (arr[minIndex],arr[i]) = (arr[i],arr[minIndex])

arr = [10,6,37,1,0,2,7]
selection_sort(arr)
print(arr)

File: Sorting/selection_sort.cpp
#include <iostream>
using namespace std;

void display(int arr[],int n){
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
}

// Time C = O(n*2)  Space C = O(1) of selection sort
// it's neither adaptive nor stable 
void selection_sort(int arr[],int n){
    for(int i = 0; i < n-1; i++){
        int minIndex = i;
        for(int j = i+1; j < n ; j++){
            if(arr[j] < arr[minIndex]) minIndex = j;
        } 
        swap(arr[minIndex],arr[i]);
    }
   display(arr,5);
}


int main()
{
    int arr[ ] = {10,9,8,7,6,};
    int n;
    selection_sort(arr,5);

    return 0;
}